integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/search/predicate/MatchSearchPredicateIT.java

```diff
@@ -922,6 +923,22 @@ public void analyzerIgnore_fuzzy() {
 				.hasDocRefHitsAnyOrder( INDEX_NAME, DOCUMENT_1, DOCUMENT_2, DOCUMENT_3 );
 	}
 
+	@Test
+	@TestForIssue( jiraKey = "HSEARCH-2534" )
+	public void analyzerOverride_queryOnlyAnalyzer() {
+		StubMappingSearchScope scope = indexManager.createSearchScope();
+		String absoluteFieldPath = indexMapping.whitespaceLowercaseAnalyzedField.relativeFieldName;
+
+		IndexSearchQuery<DocumentReference> query = scope.query()
+				.asReference()
+				.predicate( f -> f.match().onField( absoluteFieldPath ).matching( "worldofwordcraft" )
+					.analyzer( OverrideAnalysisDefinitions.ANALYZER_NGRAM.name ) )
```

Comment:
Ok, that's a weird usage of the ngram analyzer, but if it works :man_shrugging:

Reply:
Maybe just add a first query checking that the predicate won't match anything if you don't override the analyzer? So that we're sure the analyzer was indeed applied and was useful.

Reply:
good idea having a comparison here.